cfme/tests/infrastructure/test_providers.py

```diff
@@ -492,3 +495,38 @@ def test_infra_discovery_screen(appliance):
                    "to_ip4": ips['to']})
         view.start.click()
         view.flash.assert_message(ips['msg'])
+
+
+@pytest.fixture
+def setup_provider_min_templates(request, appliance, provider, min_templates):
+    templates_yaml = len(provider.data.get('templates', {}))
+    if templates_yaml < min_templates:
+        pytest.skip(f'Number of templates defined in yaml for {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    if len(provider.mgmt.list_host()) < min_templates:
+        pytest.skip(f'Number of templates on {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    # Function-scoped fixture to set up a provider
+    setup_or_skip(request, provider)
+
+
+@pytest.mark.parametrize("min_templates", [2, 4])
+@pytest.mark.meta(blockers=[BZ(1784180, forced_streams=["5.10"])], automates=[1784180])
+def test_compare_provider_templates(appliance, setup_provider_min_templates, provider,
+                                    min_templates):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    Bugzilla:
+        1746449
+    """
+    my_slice = slice(0, min_templates, None)
+    view = navigate_to(provider, 'ProviderTemplates')
+    for t in view.entities.get_all(slice=my_slice):
+        t.ensure_checked()
+    view.toolbar.configuration.item_select('Compare Selected Templates', handle_alert=True)
+    compare_templates_view = provider.create_view(TemplatesCompareView)
+    assert compare_templates_view.is_displayed
```

Comment:
Do you want to also check that the templates displayed are the ones checked? I'd consider this optional.

Reply:
Agreed on both accounts @john-dupuy - this should be do-able with the `BaseNonInteractiveEntitiesView` on `TemplatesCompareView`, and I'd like to see that implemented as collection method (the selection+widget interaction) that returns a View instance.  Optional for this PR though, as this is sufficient for BZ coverage.

Reply:
I am having trouble understanding ```implemented as collection method (the selection+widget interaction) that returns a View instance```

Reply:
I added code to check that the items checked for comparison appear in the TemplatesCompareView. Future PRs will wrap the compare functionality into code that can be used for other entities.